torch_frame/data/multi_nested_tensor.py

```diff
@@ -0,0 +1,197 @@
+
+        Returns:
+            MultiNestedTensor: Returned class object.
+        """
+        num_rows = len(tensor_mat)
+        num_cols = len(tensor_mat[0])
+
+        offset_list = []
+        accum_idx = 0
+        offset_list.append(accum_idx)
+        values_list = []
+        for i in range(num_rows):
+            for j in range(num_cols):
+                tensor = tensor_mat[i][j]
+                if not isinstance(tensor, Tensor):
+                    raise RuntimeError(
+                        "The element of tensor_mat must be PyTorch Tensor")
+                values_list.append(tensor)
+                accum_idx += len(tensor)
+                offset_list.append(accum_idx)
+
+        values = torch.cat(values_list)
+        offset = torch.tensor(offset_list, dtype=torch.long)
+
+        return cls(num_rows, num_cols, values, offset)
+
+    def __repr__(self) -> str:
+        name = self.__class__.__name__
+        name += f"(num_rows: {self.num_rows}, num_cols: {self.num_cols})"
+        return name
+
+    def __copy__(self) -> 'MultiNestedTensor':
+        out = self.__class__.__new__(self.__class__)
+        for key, value in self.__dict__.items():
+            out.__dict__[key] = value
+
+        out.values = copy.copy(out.values)
+        out.offset = copy.copy(out.offset)
+
+        return out
+
+    def __getitem__(
+        self,
+        index: Any,
+    ) -> Union['MultiNestedTensor', Tensor]:
+        if isinstance(index, Tuple):
+            # Get an element of (i, j). Returns Tensor
+            assert len(index) == 2
+            if index[0] < 0 or index[0] >= self.num_rows:
+                raise IndexError(
+                    f"Index out-of-bounds. The first element of {index} needs "
+                    f"to be [0, {self.num_rows}]")
+            if index[1] < 0 or index[1] >= self.num_cols:
+                raise IndexError(
+                    f"Index out-of-bounds. The second element of {index} "
+                    f"needs to be [0, {self.num_cols}]")
+            idx = index[0] * self.num_cols + index[1]
+            start_idx = self.offset[idx]
+            end_idx = self.offset[idx + 1]
+            out = self.values[start_idx:end_idx]
+            return out
+        elif isinstance(index, int):
+            if index < 0 or index >= self.num_rows:
+                raise IndexError(
+                    f"Index out-of-bounds. The {index} needs to be "
+                    f"[0, {self.num_rows}]")
+            # Get i-th row. Returns MultiNestedTensor
+            start_idx = index * self.num_cols
+            end_idx = (index + 1) * self.num_cols + 1
+            offset = self.offset[start_idx:end_idx]
+            values = self.values[offset[0]:offset[-1]]
+            offset = offset - offset[0]
+            return MultiNestedTensor(num_rows=1, num_cols=self.num_cols,
+                                     values=values, offset=offset)
+        elif isinstance(index, Tensor) and index.ndim == 1:
```

Comment:
we should just expose `index_select`, just as PyTorch tensors do.

Reply:
Sounds good.